### ISSUE TITLE
Reconnect to Protect controller not working

### ISSUE THREAD

**Issue by denniedegroot** [steffjenl/com.ubnt.unifiprotect#22]
When the Protect controller is restarted/rebooted the app will not reconnect.
I think it is because of the following line:
https://github.com/steffjenl/com.ubnt.unifiprotect/blob/75cf38e7bb4d66a0cd0455472bc921c885f19540/library/websocket.js#L133

It probably should be: ```setTimeout(this.waitForBootstrap().bind(this), 250);```
I haven't tested this, so please test it before committing.

**Comment by steffjenl** [comment]
> When the Protect controller is restarted/rebooted the app will not reconnect.
> I think it is because of the following line:
> 
> https://github.com/steffjenl/com.ubnt.unifiprotect/blob/75cf38e7bb4d66a0cd0455472bc921c885f19540/library/websocket.js#L133
> 
> It probably should be: `setTimeout(this.waitForBootstrap().bind(this), 250);`
> I haven't tested this, so please test it before committing.

What is different? @denniedegroot

**Comment by denniedegroot** [comment]
Function () brackets

**Comment by steffjenl** [comment]
```php
    reconnectUpdatesListener() {
        Homey.app.debug('Called reconnectUpdatesListener');
        this.disconnectEventListener().then((res) => {
            this.waitForBootstrap();
        }).catch();
    }

    waitForBootstrap() {
        Homey.app.debug('Calling waitForBootstrap A');
        if (typeof Homey.app.api._lastUpdateId !== 'undefined' && Homey.app.api._lastUpdateId !== null && this._test) {
            Homey.app.debug('Called waitForBootstrap C');
            this.launchUpdatesListener();
            this.configureUpdatesListener(this);
        } else {
            this._test = true;
            Homey.app.debug('Calling waitForBootstrap B');
            setTimeout(this.waitForBootstrap.bind(this), 250);
        }
    }
```
![image](https://user-images.githubusercontent.com/2580852/108606602-3d552180-73bb-11eb-8223-f47b3afcdaa7.png)

setTimeOut is always without function () brackets.

**Comment by denniedegroot** [comment]
@steffjenl you are correct. I still have the issue that it is not reconnecting when the controller is restarted. It seems to connect but the socket is immediately closed. Maybe ```this.webclient.get('bootstrap')``` needs to be recalled to refresh the lastUpdateId?

Can you reproduce this? Just restart the Protect controller and watch the logs.

**Comment by steffjenl** [comment]
> @steffjenl you are correct. I still have the issue that it is not reconnecting when the controller is restarted. It seems to connect but the socket is immediately closed. Maybe `this.webclient.get('bootstrap')` needs to be recalled to refresh the lastUpdateId?
> 
> Can you reproduce this? Just restart the Protect controller and watch the logs.

Elke nacht wordt onze Homey opnieuw gestart zonder problemen? 

Er is wel een probleem met de csrf token helaas met de patch opdrachten
